src/rt/atomic.rs

```diff
@@ -311,3 +745,26 @@ fn is_seq_cst(order: Ordering) -> bool {
         _ => false,
     }
 }
+
+fn range(cnt: u16) -> (usize, usize) {
+    let start = index(cnt.saturating_sub(MAX_ATOMIC_HISTORY as u16));
+    let mut end = index(cmp::min(cnt, MAX_ATOMIC_HISTORY as u16));
+
+    if end == 0 {
+        end = MAX_ATOMIC_HISTORY;
+    }
+
+    assert!(
+        start <= end,
+        "cnt = {}; start = {}; end = {}",
+        cnt,
+        start,
+        end
+    );
```

Comment:
i'm assuming this is a loom bug, not a user error, but it would be nice to have that clearly stated in the panic message?